data/campaigns/The_South_Guard/scenarios/05_Choice_In_The_Fog.cfg

```diff
@@ -864,6 +864,7 @@
 
                             [show_objectives][/show_objectives]
 
+                            # TODO: what is the effect of these lines? CLEAR_VARIABLE cannot undo an [effect]...
```

Comment:
I guess it rebuilds the unit, thus maybe working around the broken use of TRAIT_LOYAL inside a `[modify_unit][modifications]` block.

I think #5634 fixes it properly, and that PR also takes your improvement for TSG S02 from the current PR.